hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/mint/MintTranslator.java

```diff
@@ -58,18 +63,54 @@ public boolean matches(@NonNull final HtsCallAttempt attempt) {
     public @Nullable MintCall callFrom(@NonNull final HtsCallAttempt attempt) {
         final var selector = attempt.selector();
         final Tuple call;
+        final long amount;
         if (Arrays.equals(selector, MintTranslator.MINT.selector())) {
             call = MintTranslator.MINT.decodeCall(attempt.input().toArrayUnsafe());
+            amount = ((BigInteger) call.get(1)).longValueExact();
         } else {
             call = MintTranslator.MINT_V2.decodeCall(attempt.input().toArrayUnsafe());
+            amount = call.get(1);
         }
         final var token = attempt.linkedToken(Address.fromHexString(call.get(0).toString()));
         if (token == null) {
             return null;
         } else {
             return token.tokenType() == TokenType.FUNGIBLE_COMMON
-                    ? new FungibleMintCall(attempt.enhancement())
-                    : new NonFungibleMintCall(attempt.enhancement());
+                    ? fungibleCallFrom(attempt.senderAddress(), call.get(0), amount, attempt)
+                    : nonFungibleCallFrom(attempt.senderAddress(), call.get(0), call.get(2), attempt);
         }
     }
+
+    private FungibleMintCall fungibleCallFrom(
+            @NonNull final org.hyperledger.besu.datatypes.Address sender,
+            @NonNull final com.esaulpaugh.headlong.abi.Address token,
+            final long amount,
+            @NonNull final HtsCallAttempt attempt) {
+        return new FungibleMintCall(
+                attempt.enhancement(),
+                amount,
+                ConversionUtils.asTokenId(token),
+                attempt.defaultVerificationStrategy(),
+                sender,
+                attempt.addressIdConverter());
+    }
+
+    private NonFungibleMintCall nonFungibleCallFrom(
+            @NonNull final org.hyperledger.besu.datatypes.Address sender,
```

Comment:
again we can remove `org.hyperledger.besu.datatypes.` and leave it just `Address` as we have the import on top